contrib/olivere/elastic/elastictrace.go

```diff
@@ -48,15 +49,13 @@ func (t *httpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	)
 	defer span.Finish()
 
-	contentLength, err := strconv.Atoi(req.Header.Get("Content-Length"))
-	if req.Body != nil && err != nil && contentLength < maxContentLength {
-		buf, err := ioutil.ReadAll(req.Body)
-		req.Body.Close()
+	if req.Body != nil {
+		buf := bufio.NewReaderSize(req.Body, maxContentLength)
+		snapshot, err := buf.Peek(maxContentLength)
 		if err != nil {
 			return nil, err
 		}
-		span.SetTag("elasticsearch.body", string(buf))
-		req.Body = ioutil.NopCloser(bytes.NewBuffer(buf))
```

Comment:
The documentation of `bufio.Reader` is confusing. Internally, it _does_ advance the reader, so we need to do this here:
```go
req.Body = ioutil.NopCloser(buf)
```
Otherwise whatever we "peeked" at will not be forwarded.

Reply:
Good to know, I will fix this.